src/core/binding.c

```diff
@@ -58,7 +58,9 @@ QuicBindingInitialize(
     Binding->RefCount = 0; // No refs until it's added to the library's list
     Binding->Exclusive = !(UdpConfig->Flags & CXPLAT_SOCKET_FLAG_SHARE);
     Binding->ServerOwned = !!(UdpConfig->Flags & CXPLAT_SOCKET_SERVER_OWNED);
-    Binding->Connected = UdpConfig->RemoteAddress == NULL ? FALSE : TRUE;
+    Binding->Connected =
+        (QuicAddrGetFamily(&UdpConfig->Route->RemoteAddress) == QUIC_ADDRESS_FAMILY_UNSPEC) ?
```

Comment:
Isn't UNSPEC generally allowed for remote address? Like if you don't care if its v4 or v6. Or is that already fixed up by the time it gets here?

Reply:
No, we wouldn't know what address to connect to if UNSPEC was passed